source/common/http/filter_manager.h

```diff
@@ -893,6 +916,11 @@ class FilterManager : public ScopeTrackedObject,
     bool decoder_filter_chain_aborted_ : 1;
     bool encoder_filter_chain_aborted_ : 1;
     bool saw_downstream_reset_ : 1;
+    bool stream_closed_ : 1; // Set when both remote_decode_complete_ and remote_encode_complete_ is
+                             // true observed for the first time and prevents ending the stream
+                             // multiple times. Only set when allow_upstream_half_close is enabled.
+    bool should_stop_decoding_ : 1; // Set to indicate that decoding on the stream should be stopped
```

Comment:
this is what local complete was used for previously.  And it appears to me in most of the decode calls we abort if local complete and not based on should_stop_decoding which makes me think this is not doing what it says it's doing.  I think we're adding complexity and not correctness here,